### DISCUSSION TITLE
New account registration with API

### DISCUSSION THREAD

**Discussion by helpobras** [chatwoot/chatwoot#2821] [Self hosted] [answered]
Friends, I'm new to using chatwoot and I need some help with the API. I'm trying to create a new account and I always get the answer:
{
     "message": "Name can't be blank, Name is too short (minimum is 1 character)"
}
I'm using the following parameters:

{
"account_name": "My Name",
     "name":"ts informatica",
"email": "my_email@hotmail.com",
     "password": "password",
"password_confirmation": "password"
}

What is missing?

**Comment by sojan-official** [comment]
could you post the endpoint you are hitting against?

**Reply by helpobras** [reply]
I'm sorry..I went to edit and ended up deleting the post:
with postman I made the following call based on a chatwoot API I found on the internet.
#####################
$curl = curl_init();

curl_setopt_array($curl, array(
   CURLOPT_URL => 'https://dominio.com.br/api/v1/accounts',
   CURLOPT_RETURNTRANSFER => true,
   CURLOPT_ENCODING => '',
   CURLOPT_MAXREDIRS => 10,
   CURLOPT_TIMEOUT => 0,
   CURLOPT_FOLLOWLOCATION => false,
   CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
   CURLOPT_CUSTOMREQUEST => 'POST',
   CURLOPT_POSTFIELDS =>'{
"account_name": "Jean Matos",
     "name":"ts info",
"email": "jean.mcp@hotmail.com",
     "password": "xxxxx",
"password_confirmation": "xxxxx"
}',
   CURLOPT_HTTPHEADER => array(
     'Content-Type: application/json'
   ),
));

$response = curl_exec($curl);

curl_close($curl);
echo $response;
####################

**Comment by sojan-official** [comment, marked as answer]
@helpobras replace the url with your installation URL and params with required values

## curl
```
curl --location --request POST 'http://localhost:3000/api/v1/accounts' \
--header 'Content-Type: application/json' \
--data-raw '{
   "email": "test@test.com",
   "account_name": "Test Account",
   "user_full_name": "Test User",
   "password": "Password1!"
}'
```
## php 
```
<?php

$curl = curl_init();

curl_setopt_array($curl, array(
  CURLOPT_URL => 'http://localhost:3000/api/v1/accounts',
  CURLOPT_RETURNTRANSFER => true,
  CURLOPT_ENCODING => '',
  CURLOPT_MAXREDIRS => 10,
  CURLOPT_TIMEOUT => 0,
  CURLOPT_FOLLOWLOCATION => true,
  CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
  CURLOPT_CUSTOMREQUEST => 'POST',
  CURLOPT_POSTFIELDS =>'{
   "email": "test@test.com",
   "account_name": "Test Account",
   "user_full_name": "Test User",
   "password": "Password1!"
}',
  CURLOPT_HTTPHEADER => array(
    'Content-Type: application/json'
  ),
));

$response = curl_exec($curl);

curl_close($curl);
echo $response;
```

**Comment by helpobras** [comment]
wonderful. worked, thank you.

**Comment by mmojadad** [comment]
Chatwoot version : v2.11.0
Environment : Docker
All chat conversation transcript emails deliver correctly
All Agent confirmation emails deliver correctly
Issue just occur when i try create a new Account via API, No email will send.
Question : How and if is it possible to Call skip_confirmation! via API ? If yes, What will be parameters?

**Reply by dantemessy** [reply]
@mmojadad I am using v2.17.1, and to skip confirmation emails, I am creating agents using the platform APIs
